eduhelx_jupyterlab_prof/handlers.py

```diff
@@ -211,118 +130,259 @@ async def get_value(self, current_path: str):
             self.set_status(400)
             self.finish(json.dumps({
                 "message": "Not in a git repository"
             }))
             return
-        except NotStudentClassRepositoryException:
+        except NotInstructorClassRepositoryException:
             self.set_status(400)
             self.finish(json.dumps({
-                "message": "Not in student's class repository"
+                "message": "Not in your class repository"
             }))
             return
         
-        if student_repo.current_assignment is None:
+        if instructor_repo.current_assignment is None:
             self.set_status(400)
             self.finish(json.dumps({
                 "message": "Not in an assignment directory"
             }))
             return
 
-        current_assignment_path = student_repo.get_assignment_path(student_repo.current_assignment)
-        stage_files(".", path=student_repo.repo_root)
-        commit_id = commit(
-            submission_summary,
-            submission_description if submission_description else None,
-            path=student_repo.repo_root
-        )
-        push("origin", "master", path=student_repo.repo_root)
+        current_assignment_path = instructor_repo.get_assignment_path(instructor_repo.current_assignment)
+
+        rollback_id = get_head_commit_id(path=instructor_repo.repo_root)
+        stage_files(".", path=current_assignment_path)
+        
         try:
-            await self.api.create_submission(
-                student_repo.current_assignment["id"],
-                commit_id
+            commit_id = commit(
+                submission_summary,
+                None,
+                path=current_assignment_path
             )
+        except Exception as e:
+            # If the commit fails, reset and abort.
+            git_reset(".", path=current_assignment_path)
+            self.set_status(500)
+            self.finish(str(e))
+            return
+        
+        try:
+            push(ORIGIN_REMOTE_NAME, MAIN_BRANCH_NAME, path=current_assignment_path)
             self.finish()
-        except requests.exceptions.HTTPError as e:
-            self.set_status(e.response.status_code)
-            self.finish(e.response.text)
+        except Exception as e:
+            # If the push fails, but we've already committed,
+            # rollback the commit and abort.
+            git_reset(rollback_id, path=instructor_repo.repo_root)
+            self.set_status(500)
+            self.finish(str(e))
+
+class SyncToLMSHandler(BaseHandler):
+    @tornado.web.authenticated
+    async def post(self):
+        await self.api.sync_to_lms()
+        self.finish()
+
+class GradeAssignmentHandler(BaseHandler):
+    # assignment_id -> job id
+    GRADING_JOBS = {}
+
+    @tornado.web.authenticated
+    async def put(self):
+        data = self.get_json_body()
+        assignment_id: int = data["assignment_id"]
+
+        # grades = [row for row in csv.DictReader(
```

Comment:
If you mean to leave this as commented, add a comment above it describing why it's here so we don't delete it later during cleanup.

Reply:
Grading isn't implemented yet